_posts/2022-09-27-building-container-images.adoc

```diff
@@ -0,0 +1,173 @@
+== STEP 1: Creating an OpenShift Security Context Constraint [[step_1]]
+
+In OpenShift, permissions for Pods are controlled with link:https://docs.openshift.com/container-platform/latest/authentication/managing-security-context-constraints.html[security context constraints (SCC)]. OpenShift includes some pre-defined SCCs but the `restricted` SCC (the default one) does not provide enough capabilities and the `non-root` SCC provides more capabilities than required. To be able to build containers but avoid granting unrequired privileges we need to define an ad-hoc SCC that we call `container-build`. 
+
+Use an admin account to create it on an OpenShift cluster with the following command:
+
+[source,bash]
+----
+kubectl apply -f - <<EOF
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: container-build
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
+allowPrivilegedContainer: false
+allowedCapabilities:
+  - SETUID
+  - SETGID
+defaultAddCapabilities: null
+fsGroup:
+  type: MustRunAs
+# Temporary workaround for https://github.com/devfile/devworkspace-operator/issues/884
+priority: 20
+readOnlyRootFilesystem: false
+requiredDropCapabilities:
+  - KILL
+  - MKNOD
+runAsUser:
+  type: MustRunAsRange
+seLinuxContext:
+  type: MustRunAs
+supplementalGroups:
+  type: RunAsAny
+users: []
+groups: []
+volumes:
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - persistentVolumeClaim
+  - projected
+  - secret
+EOF
+----
+
+== STEP 2: Grant privileges to the DevWorkspace controller Service Account [[step_2]]
+
+The {devworkspace} controller provisions Che workspaces Pods and it uses Service Account `system:serviceaccount:openshift-operators:devworkspace-controller-serviceaccount`. 
+
+Execute the following commands to grant to the Service Account `get` and `update` privileges on the `container-build` SCC:
+
+[source, bash]
+----
+# Create the cluster role get-n-update-container-build-scc
+kubectl apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: get-n-update-container-build-scc
+rules:
+- apiGroups:
+  - "security.openshift.io"
+  resources:
+  - "securitycontextconstraints"
+  resourceNames:
+  - "container-build"
+  verbs:
+  - "get"
+  - "update"
+EOF
+
+# Add the role to the DevWorkspace controller Service Account
+oc adm policy add-cluster-role-to-user \
+       get-n-update-container-build-scc \
+       system:serviceaccount:openshift-operators:devworkspace-controller-serviceaccount
+----
+
+== STEP 3: Grant privileges to developers accounts [[step_3]]
+
+To avoid a privilege escalation, when provisioning the workspace Pod, the {devworkspace} controller checks that the developer is allowed to use the `container-buld` SCC. An administrator needs to grant such privileges. Here is an example of the command to add the `container-build` SCC to the user `janedoe`:
```

Comment:
```suggestion
To avoid a privilege escalation, when provisioning the workspace Pod, the {devworkspace} controller checks that the developer is allowed to use the `container-build` SCC. An administrator needs to grant such privileges. Here is an example of the command to add the `container-build` SCC to the user `janedoe`:
```